app/assets/javascripts/proceeding_types.js

```diff
@@ -1,56 +1,45 @@
-var proceeding_types = {
-  init: function(){
-    this.setup_proceeding_search();
-    this.setup_clear_search();
-  },
-  setup_proceeding_search: function(){
-    var minimum_char_length = 3;
-
-    $('#proceeding-search').on('keyup', function (e) {
-      var current_input = $(this).val().toLowerCase();
-      if (current_input.length >= minimum_char_length) {
-        $.getJSON("/v1/proceeding_types", function (proceedings_data) {
-          var filtered_proceedings = $.grep(proceedings_data, function (proceeding) {
-            return (proceeding.description.toLowerCase().indexOf(current_input) !== -1);
-          });
-
-          $("#proceeding-list").empty();
-            if (filtered_proceedings.length !== 0) {
-              $.each(filtered_proceedings, function (_, proceeding) {
-                var proceeding_row =  templates.proceeding_result_template(
-                                        proceeding.category_law,
-                                        proceeding.matter,
-                                        proceeding.meaning,
-                                        proceeding.code
-                                      );
-
-                $("#proceeding-list").append(proceeding_row);
-              });
-            } else {
-              var proceeding_row = templates.no_results_template;
-              $("#proceeding-list").append(proceeding_row);
-            }
+$.getJSON("/v1/proceeding_types", function (proceedings_data) {
+
+  $("#proceeding-list .proceeding-item").hide();
+  $(".no-proceeding-items").hide();
+
+  // For docs see: https://github.com/bvaughn/js-search
+  let search = new JsSearch.Search("code");
+  search.addIndex("meaning");
+  search.addIndex("description");
+  search.addIndex("category_law");
+  search.addIndex("matter");
+  search.addDocuments(proceedings_data);
+
+  $("#proceeding-search-input").keyup(function(){
+    $("#proceeding-list .proceeding-item").hide();
+    $(".no-proceeding-items").hide();
+
+    // Get user input and filter on it.
+    let inputText = $(this).val();
+
+    if (inputText.length > 2) {
+      let codes = search.search(inputText).map(function(obj) {return obj["code"];});
+
+      if (codes.length > 0) {
+        // Iterate through each code, find the matching element, move it to the
+        // top and display it
+        $.each(codes.reverse(), function(_i, code){
+          let element = $('#' + code);
+          let parent = element.parent("div");
```

Comment:
**[not a blocker]** For more reliability, consider defining the ID of the object to prepend the _element_ to as a **data** attribute (this avoids trying to navigate through the DOM and just _link_ the two objects semantically), e.g:

```ruby
# proceeding type partial view
<div id="<%= proceeding_type.code %>" class="govuk-grid-row proceeding-item" data-list-container="proceeding-list">
...
</div>

```

```js
# this JS file
let element = $('#' + code);
let container = element.data("list-container");
element.detach().prependTo(container);
```